[Docs] Guide on setting query context (#18179)
apache/druid: docs/querying/sql-query-context.md

Reviewer: Remove section on `## Set the query context` since it is now moved to the new doc (which you linked to in the introduction)

:::

> Druid supports query context parameters which affect [SQL query](./sql.md) planning.

Reviewer: Suggest putting together a more coherent introduction:

Ex. paragraph 1
* what are they
* (what does that mean)
* link to how to set it
* link to reference page

Ex. paragraph 2: "you can override"

Ex. paragraph 3: "the following table lists"

:::

Druid supports query context parameters which affect [SQL query](./sql.md) planning.
See [Query context](query-context-reference.md) for general query context parameters for all query types.

> For more detailed instructions on how to set query context in Apache Druid, see [Set the query context](../querying/set-query-context.md)

Reviewer: ```suggestion
For information on how to set the query context, see [Set query context](../querying/set-query-context.md)
```

|`inFunctionExprThreshold`|At or beyond this threshold number of values, SQL `IN` is eligible for execution using the native function `scalar_in_array` rather than an <code>&#124;&#124;</code> of `==`, even if the number of values is below `inFunctionThreshold`. This property only affects translation of SQL `IN` to a [native expression](math-expr.md). It doesn't affect translation of SQL `IN` to a [native filter](filters.md). This property is provided for backwards compatibility purposes, and may be removed in a future release.|`2`|
|`inSubQueryThreshold`|At or beyond this threshold number of values, Druid converts SQL `IN` to `JOIN` on an inline table. `inFunctionThreshold` takes priority over this setting. A threshold of 0 forces usage of an inline table in all cases where the size of a SQL `IN` is larger than `inFunctionThreshold`. A threshold of `2147483647` disables the rewrite of SQL `IN` to `JOIN`. |`2147483647`|

## Learn more

> - [Set query context](../querying/set-query-context.md) for how to set query context.

Reviewer: "how to set **the** query context"
